cyrasterize/glrasterizer.pyx

```diff
@@ -104,6 +104,32 @@ def normalize_v3(arr):
     arr[:,2] /= lens
     return arr
 
+
+def vertex_normals(points, trilist):
+    # Create a zeroed array with the same type and shape as our vertices i.e., per vertex normal
+    cdef np.ndarray[float, ndim=2, mode="c"] norm = np.zeros( (points.shape[0], points.shape[1]), dtype=np.float32)
+    tris = points[trilist]
+
+    # Calculate the normal for all the triangles, by taking the cross product of the vectors v1-v0, and v2-v0 in each triangle
+    n = np.cross( tris[::,1 ] - tris[::,0]  , tris[::,2 ] - tris[::,0] )
```

Comment:
sorry @patricksnape missed this!

Yeah I know, could be brought over for sure.

To be clear this only moves the existing normals calc into a function, did the timings and our vertex normals  calc is faster for sure. For now just focused on getting this together for release so we can make progress with the 3DMM project.